fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -645,4 +725,25 @@ private void checkLinkForLdpResourceCreation(final String link) {
         }
     }
 
+    /**
+     * Parse the RFC-3230 Digest response header value.  Look for a
+     * sha1 checksum and return it as a urn, if missing an empty string is
+     * returned.
+     * @param digest The Digest header value
+     * @return the sha1 checksum value
+     */
+    private String parseDigestHeader(final String digest) {
+        if (digest != null) {
+            final Map<String,String> digestPairs = RFC3230_SPLITTER.split(digest);
+            for (Map.Entry<String, String> digestPair : digestPairs.entrySet()) {
+                switch(digestPair.getKey().toLowerCase()) {
```

Comment:
I think it would be great to support other checksums.  We could calculate other checksums and check them too.  And we could also add the additional checksums as properties on the binary.  So I'm fine with leaving some of the framework in place for handling additional checksum algorithms, and adding a new ticket for adding that support.